docs/design/architectural-overview.md

```diff
@@ -0,0 +1,168 @@
+
+
+
+
+### Data Flows
+
+There are several different data flows when using Kuadrant. 
+
+**Control plane configuration and status reporting**
+
+The initial creation of these APIs (gateways, policies etc) is done by the relevant persona in the control plane just as they would any other k8s resource. We use the term cluster admin or gateway admin as the operatoins type persona configuring, and placing gateways. 
+As shown above, in a multi-cluster configuration. API definitions are pulled from the Hub and "manifested" into the spokes. The Status of those synced resources are reported back to the Hub. The same happens for a single cluster, the only difference being the work agent hub controllers are all installed on one cluster.
+
+**3rd party enforcement and Integration**
+
+In order to enforce the policy configuration, components in the control plane and data plane can reach out to configured 3rd parties such as cloud based DNS provider, TLS providers and Auth providers.
+
+
+**Request Flow**
+
+Requests coming through the gateway instance are intercepted by the WASM plugin installed into the Envoy based gateway provider. This plugin is then responsible for passing on the request to Limitador to be rate limited or Authorino to ensure auth. Each of these component each have the capability to see the request and need to in order to make the required decision. Each of these component can also prevent the request from reaching its backend destination based on user configuration. 
+
+
+![](./images/request-flow.png)
+
+
+## Auth
+
+Authentication is handled by the underly kubernetes cluster. All APIs are CRDs and so authorization is controlled by RBAC in the same way as any kubernetes API.
+
+For Auth between Spoke and Hub see OCM docs https://open-cluster-management.io/concepts/architecture/
+
+
+
+### Observability
+
+Kuadrant doesn't provide any specific observability components, but rather provides a reference setup using well known and established components along with some useful dashboards to help observe key things around the Gateways. The focus of this setup, is in the context of a multi-cluster setup where OCM is installed and gateways are being defined and distributed from that hub.
+
+![](./images/observability.png)
+
+
+## Some notes on future direction
+
+This section is here to provide some insight into architectural changes that may be seen in the near future:
+
+What is in this doc represents the architecture at point our MVP release. Below are some areas that we have identified that are likely to change in the coming releases. As these happen, this doc will also evolve.
+
+
+- We want to separate out the ocm integration into its own controller
```

Comment:
```suggestion
- We want to separate out the ocm integration into its own controller so that policies can evolve without a coupling to any one multi-cluster management solution
```
I think this is the reason for this future direction?